src/org/parosproxy/paros/core/scanner/VariantMultipartFormParameters.java

```diff
@@ -0,0 +1,189 @@
+ * all parameters passed inside the form-data structure and sets them for 
+ * injection.
+ */
+public class VariantMultipartFormParameters implements Variant {
+
+	private static final Logger LOGGER = Logger.getLogger(VariantMultipartFormParameters.class);
+	private static final Pattern FIELD_NAME_PATTERN = Pattern.compile(
+			"\\s*content-disposition\\s*:.*\\s+name\\s*\\=?\\s*\\\"?(?<name>.[^;\\\"\\n]*)\\\"?\\;?.*",
+			Pattern.CASE_INSENSITIVE);
+	private static final Pattern FIELD_VALUE_PATTERN = Pattern.compile("[\\r\\n]{2}(?<value>.*)");
+	private static final Pattern FILENAME_PART_PATTERN = Pattern.compile(
+			"\\s*content-disposition\\s*:.*filename\\s*\\=?\\s*\\\"?(?<filename>.[^;\"\\n]*)\\\"?\\;?.*",
+			Pattern.CASE_INSENSITIVE);
+			// http://fiddle.re/etxbnd (Click Java, set case insensitive, and hit "test")
+	private static final Pattern CONTENTTYPE_PART_PATTERN = Pattern.compile(
+			"\\s*content-disposition.*content-type\\s*:\\s*\\s*\\\"?(?<contenttype>.[^;\"\\r\\n]*)\\\"?\\;?.*",
+			Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
+			// http://www.regexplanet.com/share/index.html?share=yyyyyythear (Click Java, set case 
+			// insensitive & DOTALL, and hit "test")
+
+	private List<NameValuePair> params = Collections.emptyList();
+	private List<MultipartFormParameter> multiPartParams = new ArrayList<MultipartFormParameter>();
+
+	@Override
+	public void setMessage(HttpMessage msg) {
+		if (msg == null) {
+			throw new IllegalArgumentException("Parameter message must not be null.");
+		}
+
+		String contentType = msg.getRequestHeader().getHeader(HttpHeader.CONTENT_TYPE);
+		if (contentType == null || !contentType.toLowerCase().startsWith("multipart/form-data")) {
+			return;
+		}
+
+		ArrayList<NameValuePair> extractedParameters = new ArrayList<>();
+		int position = 0;
+		int offset = 0;
+		String boundary = getBoundary(contentType) + HttpHeader.CRLF;
+		for (String part : msg.getRequestBody().toString().split(boundary)) {
+			if (!StringUtils.isBlank(part)) {
+				String partHeaderLine = part.substring(0, part.indexOf(HttpHeader.CRLF + HttpHeader.CRLF));
+				part = boundary + part;
+				Matcher nameMatcher = FIELD_NAME_PATTERN.matcher(partHeaderLine);
+				Matcher valueMatcher = FIELD_VALUE_PATTERN.matcher(part);
+				nameMatcher.find();
+				valueMatcher.find();
+				if (StringUtils.isBlank(valueMatcher.group("value"))) {
+					valueMatcher.find();
+					// Need to skip one find for some reason...
+					// https://regex101.com/r/4ig6Wk/1
+					// http://fiddle.re/23cudd (Click Java, hit "test")
+				}
+
+				String name = nameMatcher.group("name");
+				// Value doesn't include boundary, headerline, or double CRLF
+				String value = part.replaceAll(boundary + partHeaderLine + HttpHeader.CRLF + HttpHeader.CRLF, "");
+				value = value.replaceAll(getBoundary(contentType) + "--" + HttpHeader.CRLF, "");// Strip final boundary
+				value = value.trim();// Trailing CRLF

```

Comment:
This might remove more than needed (leading and trailing), not be a big issue though.

Reply:
Hmmmm it proved more reliable than reducing the end mark by `HttpHeader.CRLF.length()`, guess I’ll have to debug the issue(s) with that further or see if StringUtils has a trimTrailing or similar.

Reply:
Right, we should be changing the `value` instead of removing the chars from the end. Using:
```Java
value = value.replaceAll(HttpHeader.CRLF + "(" + Pattern.quote(getBoundary(contentType)) + "--" + HttpHeader.CRLF + ")?$", ""); // Strip final boundary
```
seems to work in all cases.

Reply:
Thanks @thc202